src/components/Graph/ThresholdLabel.js

```diff
@@ -1,24 +1,54 @@
-import React from 'react';
+import React, { Component } from 'react';
+import TooltipHandler from '../Filters/TooltipHandler';
 import { addCommas, getReadableDate } from '../../utils/utils.js';
 
-function ThresholdLabel(props) {
-    const { r0full, r0selected } = props;
-    const chance = Math.round(100 * props.percExceedence);
-    const val = addCommas(Math.ceil(props.statThreshold / 100) * 100);
-    const date = getReadableDate(props.dateThreshold);
+class ThresholdLabel extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            showTooltip: false
+        }
+    }
+
+    handleTooltipClick = () => {
+        this.setState({showTooltip: !this.state.showTooltip})
+    }
+
+    render() {
+        const chance = Math.round(100 * this.props.percExceedence);
+        const val = addCommas(Math.ceil(this.props.statThreshold / 100) * 100);
+        const date = getReadableDate(this.props.dateThreshold);
```

Comment:
In general, you'll want to avoid processing logic in the render() function if possible. Depending on how many times render() gets called, each one of the computations will be run.

Reply:
Where should the processing logic live instead? I could create a function that's like `calc()` which is called on both `componentDidMount` for when it's mounted and `componentDidUpdate` for when the props change?

Reply:
You could set local state and update it in `componentDidUpdate`, that way it only gets calculated in specific conditions in `componentDidUpdate`.  Then you would use local state (which is already computed) in render()